### PR TITLE
Docs: Switch code distinction to JSX & Plain

### LINKED ISSUE #35776: Docs: Rename code example toggles to better represent current state
The code examples in the developer documentation, as well as the [gutenberg-examples repo](https://github.com/WordPress/gutenberg-examples), use a delineation between two types of code: ES5 and ESNext 

With the dropping of support for IE11, the targeted browsers all support the ES6 syntax. So the distinction is no longer accurate, as it is not between ES5 and a newer syntax; but between not building JavaScript and using JSX that requires a build.

So I think we should rename the toggles and our usage within documentation to better reflect this distinction. 

My suggestion is to name the toggles: "JSX" and "Plain"

We discussed this also in the [core-js meeting today](https://wordpress.slack.com/archives/C5UNMSU4R/p1634658327194600), and it was also discussed [in this ticket](https://github.com/WordPress/gutenberg/issues/33263). 

Changing the toggles will also clarify whether we need two sets of examples, as there has been [some discussion around dropping ES5 support in docs](https://github.com/WordPress/gutenberg/issues/32430). 

For reference, [our documentation around code examples](https://developer.wordpress.org/block-editor/contributors/documentation/#code-examples) recommends including ES5 examples for beginner tutorials but not required for all code examples.

Also for reference, looking to see how other documentation manages, [in the React tutorial](https://reactjs.org/tutorial/tutorial.html) they show one example using non-JSX and the rest of their documentation uses JSX. For Preact, [in the getting started guide](https://preactjs.com/guide/v10/getting-started#alternatives-to-jsx) they show an example using a JSX alternate (HTM), but then the rest of the documentation uses JSX.

So an alternative, which I'm not necessarily for but I'll include for this discussion to acknowledge it as an option, is to remove the code tabs and add a new section explaining JSX and build.

What do y'all think?

### PR DESCRIPTION
**Pull request by mkaz**
## Description

The distinction between JavaScript examples shown in code is no longer between JavaScript syntaxes, ESNext and ES5, since we dropped IE11 support; but between code that requires a build due to JSX and Plain JavaScript that does not require a build.

- Update codetabs changing lables from ESNext & ES5 to JSX & Plain.

- Updates various pages and descriptions of what version to show, including documentation guide.

- Updates block-transforms docs to remove ES5 code examples. The code examples are duplicated with the basic differences being destructuring and function definitions, no JSX code used. So with the dropping of IE11 support, our minimum targeted browser supports the ES6 code that is being used, so we don't need both examples.

## How has this been tested?

Documentation changes, check for:

- Clarity: content makes sense and accurate,
- Mistakes: Typos or spelling issues
- Complete: Anything missing?

**Note:** I did not change the examples code that use `-esnext` suffix, those changes will probably need to be coordinated with the gutenberg-examples repo and make sure any links are also updated.

## Types of changes

Documentation updates.

Closes #35776

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
This is awesome. I like how the new strategy allows removing several duplicated examples of the same code written in ESNext and ES5 which are now identical. In fact, as long as we don't deal with React components you don't need to have two versions of the same example anymore which removes a lot of overhead for both the documentation authors and the consumers of the handbook.

Great work wrangling all the changes. I haven't verified very carefully the individual examples because it's too many of them, but I verified all changes to how we present JSX and Plain version of code examples and it looks great from my perspective.

File: docs/contributors/documentation/README.md [hunk 1]
```diff
@@ -88,26 +88,26 @@ An example, the link to this page is: `/docs/contributors/documentation/README.m
...
     ```
-    {% end %}
+    \{\% end \%\}
 ````
 
-The preferred format for code examples is ESNext, this should be the default view. The example placed first in source will be shown as the default.
+The preferred format for code examples is JSX, this should be the default view. The example placed first in source will be shown as the default.
 
-**Note:** it is not required to include ES5 code examples. The guidance is to include `ES5` code for beginner tutorials, but the majority of code in Gutenberg packages and across the larger React and JavaScript ecosystem is in ESNext.
+**Note:** It is not required to include plain JavaScript code examples for every guide. The recommendation is to include plain code for beginner tutorials, or short examples; but the majority of code in Gutenberg packages and across the larger React and JavaScript ecosystem are in JSX that require builds.
```
Comment (ryanwelcher) [inline docs/contributors/documentation/README.md]:
```suggestion
**Note:** It is not required to include plain JavaScript code examples for every guide. The recommendation is to include plain code for beginner tutorials or short examples, but the majority of code in Gutenberg packages and across the larger React and JavaScript ecosystem are in JSX that requires a build process.
```

File: docs/how-to-guides/block-tutorial/README.md [hunk 2]
```diff
@@ -4,6 +4,6 @@ The purpose of this tutorial is to step through the fundamentals of creating a n
...
+Code snippets are provided in two formats "JSX" and "Plain". JSX refers to JavaScript code that uses JSX syntax which requires a build step. Plain refers to "classic" JavaScript that does not require building. You can change between them using tabs found above each code example. Using JSX, does require you to run [the JavaScript build step](/docs/how-to-guides/javascript/js-build-setup/) to compile your code to a browser compatible format.
 
-Note that it is not required to use ESNext to create blocks or extend the editor, you can use classic JavaScript. However, once familiar with ESNext, developers find it is easier to read and write, thus most code examples you'll find use the ESNext syntax.
+Note that it is not required to use JSX to create blocks or extend the editor, you can use classic JavaScript. However, once familiar with JSX and the build step, developers find it is easier to read and write, thus most code examples you'll find use the JSX syntax.
```
Comment (ryanwelcher) [inline docs/how-to-guides/block-tutorial/README.md]:
```suggestion
Note that it is not required to use JSX to create blocks or extend the editor, you can use classic JavaScript. However, once familiar with JSX and the build step, many developers tend to find it is easier to read and write, thus most code examples you'll find use the JSX syntax.
```
Softened the language as here a little, we probably shouldn't assume all developers prefer JSX over Plain

File: docs/how-to-guides/javascript/versions-and-building.md [hunk 3]
```diff
@@ -1,15 +1,14 @@
 # JavaScript Versions and Build Step
 
-The Block Editor Handbook shows JavaScript examples in two syntaxes: ES5 and ESNext. These are version names for the JavaScript language standard definitions. You may also see elsewhere the names ES6, or ECMAScript 2015 mentioned. See the [ECMAScript](https://en.wikipedia.org/wiki/ECMAScript) Wikipedia article for all the details.
+The Block Editor Handbook shows JavaScript examples in two syntaxes: JSX and Plain.
 
-ES5 code is compatible with WordPress's minimum [target for browser support](https://make.wordpress.org/core/handbook/best-practices/browser-support/).
+Plain refers to JavaScript code compatible with WordPress's minimum [target for browser support](https://make.wordpress.org/core/handbook/best-practices/browser-support/) without requiring a transpilation step.
```
Comment (ryanwelcher) [inline docs/how-to-guides/javascript/versions-and-building.md]:
```suggestion
Plain refers to JavaScript code compatible with WordPress's minimum [target for browser support](https://make.wordpress.org/core/handbook/best-practices/browser-support/) without requiring a transpilation step. This step is commonly referred to as a build process.
```
We use the phrases _build_ or _build process_ in other places to refer to this step. Just wanted to be sure we're clear that it's all the same thing.

File: docs/how-to-guides/javascript/versions-and-building.md [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
 
-"ESNext" doesn't refer to a specific version of JavaScript, but is "dynamic" and refers to the next language definitions, whatever they might be. Because some browsers won't support these features yet (because they're new or proposed), an extra build step is required to transform the code to a syntax that works in all browsers. Webpack and babel are the tools that perform this transformation step.
+"JSX" doesn't refer to a specific version of JavaScript, but refers to the latest language definition plus
+[JSX syntax](https://reactjs.org/docs/introducing-jsx.html), a syntax that blends HTML and JavaScript. JSX makes it easier to read and write markup code, but does require a build step to transform into code compatible with browers. Webpack and babel are the tools that perform this transformation step.
```
Comment (ryanwelcher) [inline docs/how-to-guides/javascript/versions-and-building.md]:
```suggestion
[JSX syntax](https://reactjs.org/docs/introducing-jsx.html), a syntax that blends HTML and JavaScript. JSX makes it easier to read and write markup code, but does require a build step to transpile into code compatible with browers. Webpack and babel are the tools that perform this transpilation step.
```

File: docs/how-to-guides/platform/custom-block-editor/README.md [hunk 5]
```diff
@@ -12,8 +12,6 @@ To follow along with this tutorial, you can [download the accompanying WordPress
 
 ## Code Syntax
 
-Code snippets are provided in "ESNext". ESNext refers to the next versions of the language standard, plus JSX syntax.
-
-Note that it is not required to use ESNext to create blocks or extend the editor, you can use classic JavaScript. However, once familiar with ESNext, developers find it is easier to read and write, thus most code examples you'll find use the ESNext syntax.
+Code snippets are provided using JSX syntax. Note it is not required to use JSX to create blocks or extend the editor, you can use plain JavaScript. However, once familiar with JSX, developers find it is easier to read and write, thus most code examples you'll find use that syntax.
```
Comment (ryanwelcher) [inline docs/how-to-guides/platform/custom-block-editor/README.md]:
```suggestion
Code snippets are provided using JSX syntax. Note it is not required to use JSX to create blocks or extend the editor, you can use plain JavaScript. However, once familiar with JSX, many developers tend find it is easier to read and write, thus most code examples you'll find use that syntax.
```

**Review by ryanwelcher — CHANGES_REQUESTED** [review]
This all looks great! I have added a few suggestions to keep our terminology consistent and to soften some language around what developers like. Just an FYI, I didn't check the code examples to ensure they work as expected.

**Review by ryanwelcher — APPROVED** [review]

**Pull request merged**